src/main/java/seedu/address/commons/core/Oauth2Client.java

```diff
@@ -75,12 +72,16 @@ public static void authenticateWithLinkedIn(Config configuration) throws IOExcep
     /**
      * Starts a webserver and allows it to expect a response at the context specified
      */
-    public static void startServer() throws IOException {
-        int port = 13370;
-        HttpServer server = HttpServer.create(new InetSocketAddress(port), 0);
-        server.createContext("/test", new MyHandler());
-        server.setExecutor(null);
-        server.start();
+    public static void startServer() {
+        try {
+            int port = 13370;
+            HttpServer server = HttpServer.create(new InetSocketAddress(port), 0);
+            server.createContext("/test", new MyHandler());
```

Comment:
I received a comment before that only the lines that are expected to fail should be wrapped in the try block. Probably not important, and I don't even know which lines are expected to fail.

Reply:
this can fail if the server has already been started by another process.